src/Sulu/Bundle/MediaBundle/Resources/public/js/components/collections/components/content/main.js

```diff
@@ -38,51 +38,41 @@ define(function() {
-                        };
-                    });
-
-                    def.resolve({
-                        tabs: {
-                            url: '/admin/media/navigation/collection'
-                        },
-                        toolbar: {
-                            template: [
+            // init locale
+            this.locale = this.sandbox.sulu.user.locale;
+
+            return {
+                tabs: {
+                    url: '/admin/media/navigation/collection'
+                },
+                toolbar: {
+                    template: [
+                        {
+                            id: 'settings',
+                            icon: 'gear',
+                            position: 30,
+                            items: [
                                 {
-                                    id: 'settings',
-                                    icon: 'gear',
-                                    position: 30,
-                                    items: [
-                                        {
-                                            id: 'collection-move',
-                                            title: this.sandbox.translate('sulu.collection.move'),
-                                            callback: this.startMoveCollectionOverlay.bind(this)
-                                        },
-                                        {
-                                            id: 'delete',
-                                            title: this.sandbox.translate('sulu.collections.delete-collection'),
-                                            callback: this.deleteCollection.bind(this)
-                                        }
-                                    ]
+                                    id: 'collection-move',
+                                    title: this.sandbox.translate('sulu.collection.move'),
+                                    callback: this.startMoveCollectionOverlay.bind(this)
+                                },
+                                {
+                                    id: 'delete',
+                                    title: this.sandbox.translate('sulu.collections.delete-collection'),
+                                    callback: this.deleteCollection.bind(this)
                                 }
-                            ],
-                            parentTemplate: 'save',
-                            languageChanger: {
-                                data: localizations,
-                                preSelected: this.options.locale
-                            }
-                        },
-                        noBack: true
-                    });
-                }.bind(this));
-
-            return def;
+                            ]
+                        }
+                    ],
+                    parentTemplate: 'save',
+                    languageChanger: {
+                        url: '/admin/api/languages',
```

Comment:
I would not use this actions, since it is in the snippet bundle (which should btw also use the `/admin/api/localizations`-action). I would suggest to rewriting the localizations action to return a correct rest response (containing `_embedded`), and use the result key here. Currently the only other bundle this has to be fixed is in the SuluSecurityBundle.